указание геолокации(без валидации)

Sergey-Ubogov opened the pull request.

Sergey-Ubogov commented on src/blocks/create-quest-page/photos-list/photos-list.js:
```diff
@@ -58,3 +72,92 @@ function makeInputs(photoBlock, index) {
+let questPhotos = block.querySelectorAll('.quest-photo');
+questPhotos.forEach(questPhoto => {
+    let btnBorder = questPhoto.querySelector('.btn-border');
+    let photoLocation = questPhoto.querySelector('.photo-location');
```
дублирование кода

AtnesNess commented on app.js:
```diff
@@ -39,6 +39,8 @@ let cachegooseConfig = {
     cachegooseConfig = require('parse-redis-url')(redis).parse(process.env.REDIS_URL);
 }
+
+//  cachegoose(mongoose);
```
что это?

Sergey-Ubogov replied:
это чтоб на локалке работало

AtnesNess replied:
для ПР это не нужно

AtnesNess commented on src/apps/quests/controllers.js:
```diff
@@ -378,20 +376,21 @@ function applyQuestData(quest, user, post, done) {
     photoTools.deletePhotos(photosToDelete, err => {
         if (err) {
-            done(err);
+            return done(err);
```
я бы не стал включать изменения которые находятся сейчас в другом ПР

AtnesNess commented on src/blocks/create-quest-page/create-quest-page.js:
```diff
@@ -1,5 +1,5 @@
 /* globals createFlashMessage */
-/* eslint no-alert:0 */
+/* eslint no-alert:0 no-undef:0 */
```
зачем?

Sergey-Ubogov replied:
я сначало здесь кодил, потом перенес код, уберу )

AtnesNess commented on src/blocks/create-quest-page/create-quest-page.less:
```diff
@@ -1,3 +1,43 @@
+}
+
+.map {
+    -moz-border-radius: 14px 14px 0 0; /* Firefox */
```
вендорные префиксы не нужно добавлять, у нас для этого gulp

AtnesNess commented on src/blocks/create-quest-page/photos-list/photos-list.js:
```diff
@@ -25,6 +30,15 @@ function imgSetSlot() {
+    let photoLocation = newPhotoBlock.querySelector('.photo-location');
+    let checkboxEdited = newPhotoBlock.querySelector('.edited');
+    btnBorder.addEventListener('click', () => {
+        popUpShadow.style.display = 'block';
```
у нас есть если что блок general-shade, он на мобилке себя норм ведет в отличае от этого =)

AtnesNess commented on src/blocks/create-quest-page/photos-list/photos-list.js:
```diff
@@ -58,3 +72,92 @@ function makeInputs(photoBlock, index) {
 };
+
+popUpShadow.addEventListener('click', e => {
+    if (e.target === popUpShadow) {
```
зачем эта проверка?

Sergey-Ubogov replied:
когда жмешь на элемент, то он рекурсивно от ребенка до родителя срабатывает, т.е. если жмешь на карту, то она сскроется

AtnesNess replied:
используй general-shade и z-index, тогда не нужно будет задумываться о родителях и всплытии

AtnesNess commented on src/blocks/create-quest-page/photos-list/photos-list.js:
```diff
@@ -1,10 +1,15 @@
 counter.value = block.querySelectorAll('.quest-photo').length - 1;
 let cloned = photoBlock.cloneNode(true);
 let dragndropInitFunction = block.querySelector('.drag-n-drop').initFunction;
+let popUpShadow = document.querySelector('.pop-up-shadow');
```
не очень хорошо когда вложенные блоки знают о том кто их в себя вложил.  Обратное возможно. когда родитель знает о своих детях. Ну да ладно пойдет

AtnesNess commented on src/blocks/create-quest-page/photos-list/photos-list.js:
```diff
@@ -58,3 +72,92 @@ function makeInputs(photoBlock, index) {
+    }
+});
+
+let btnSetGeolocation = document.querySelector('.set-geolocation');
```
тоже самое о родителях и детях

AtnesNess commented on src/blocks/create-quest-page/photos-list/photos-list.js:
```diff
@@ -58,3 +72,92 @@ function makeInputs(photoBlock, index) {
+
+        createFlashMessage('Координаты установлены!');
+    });
+    navigator.geolocation.getCurrentPosition(position => {
```
ну я думаю, у человека который редактирует должна быть возможность посмотреть куда он вообще ткнул. Т.е при открытии локации, если уже было что-то установлено нужно показывать ее, если ничего не установлено - указывать текущее местоположение.
И хорошо бы добавить кнопку, чтобы можно было перейти на текущее местоположение (как это сделано вк например)

AtnesNess reviewed (changes requested):
Выглядит неплохо, но есть недочеты:
1) Не работает на мобилке
2) Вылетает ошибка в консоль при загрузке
3) Для блока затемнения используй general-shade
4) Юзер должен понимать куда он поставил точку, после повторного открытия карты
5) Убери лишние вендорные префиксы, ненужные коменты и желательно данные из парралеьных ПРов 116 и 115
6) Если будет возможность сделай так, чтобы блоки-дети гне знали о существовании блоков-родителей. Т.е не использовать document, а только пользоваться block
7) В код клиентского js пока особо не вдавался. Исправь пока предыдущие пункты

Commit:
merge

линтер, дублирование кода

linter

правки

линтер

правки

исправил багу с тем что при установке координат, адрес был левый

document -> block, удалил лишний css код

сделал центрирование карты при установке местоположения

AtnesNess commented on src/blocks/create-quest-page/photos-list/photos-list.js:
```diff
@@ -1,10 +1,17 @@
 counter.value = block.querySelectorAll('.quest-photo').length - 1;
 let cloned = photoBlock.cloneNode(true);
 let dragndropInitFunction = block.querySelector('.drag-n-drop').initFunction;
+let popUpShadow = document.querySelector('.general-shade');
```
block вместо document

AtnesNess commented on src/blocks/create-quest-page/photos-list/photos-list.less:
```diff
@@ -11,3 +11,42 @@ input {
     display: none;
 }
+
+.pop-up-shadow {
```
это можно убрать

AtnesNess reviewed (changes requested):
Не меняется адрес(строка) при просмотри позиций разных квестов

AtnesNess commented on src/blocks/create-quest-page/photos-list/item/item.hbs:
```diff
@@ -5,6 +5,5 @@
 <input type="checkbox" class="edited" name="photo-block-{{index}}-edited"/>
 <input type="checkbox" class="deleted" name="photo-block-{{index}}-deleted"/>
 <input name="photo-block-{{index}}-photo" class="photo-link" {{bind-attr value="photo:photo"}}/>
```
может быть здесь нужен photo.url?

Commit:
вместо [Object object] в value теперь url

Commit:
добавил geoPosition там где есть cache

AtnesNess reviewed (approved):
:+1:
